source/GameData.cpp

```diff
@@ -549,21 +554,29 @@ void GameData::Change(const DataNode &node)
 
 
 
-// Update the neighbor lists of all the systems. This must be done any time
-// that a change creates or moves a system.
-void GameData::UpdateNeighbors()
+// Update the neighbor lists and other information for all the systems.
+// This must be done any time that a change creates or moves a system.
+void GameData::UpdateSystems()
 {
 	for(auto &it : systems)
 	{
 		// Skip systems that have no name.
 		if(it.first.empty() || it.second.Name().empty())
 			continue;
-		it.second.UpdateNeighbors(systems);
+		it.second.UpdateSystem(systems, neighborDistances);
 	}
 }
 
 
 
+void GameData::NeighborDistance(double neighborDistance)
+{
+	if(!neighborDistances.count(neighborDistance))
+		neighborDistances.insert(neighborDistance);
```

Comment:
This code is needed when using a vector, but when using the datatype set, then it is overkill since the check if the element is present is already done within the set itself.